java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -585,48 +585,48 @@ public ArrowBuf setDouble(int index, double value) {
   @Override
   public ArrowBuf writeShort(int value) {
     ensure(2);
-    PlatformDependent.putShort(addr(writerIndex), (short) value);
-    writerIndex += 2;
+    PlatformDependent.putShort(addr(writerIndex()), (short) value);
+    writerIndex(writerIndex() + 2);
```

Comment:
Good point. I ran this code and didn't see performance diffs, are there other tests you'd like me to run?
```
public class TestArrowBufPerformance {
    @Test
    public void test() {
        int count = 10 * 1024 * 1024;
        RootAllocator allocator = new RootAllocator(8 * count);

        for (int j = 0; j < 100; ++j) {
            ArrowBuf buf = allocator.buffer(4 * count);
            long start = System.currentTimeMillis();
            for (int i = 0; i < count; ++i) {
                buf.writeInt(i);
            }
            System.out.println(System.currentTimeMillis() - start);
            buf.close();
        }
        allocator.close();
    }
}
```

before:
```
count 	100.000000
mean 	40.650000
std 	8.048307
min 	29.000000
25% 	34.000000
50% 	38.500000
75% 	45.250000
max 	64.000000
```

after:
```
count 	100.000000
mean 	40.930000
std 	7.002532
min 	31.000000
25% 	36.000000
50% 	39.000000
75% 	47.000000
max 	65.000000
```